### PR TITLE
feat: Skeleton.List api

### PR DESCRIPTION
**Pull request by vikasprashar99**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
[https://github.com/ant-design/ant-design/issues/20574](url)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Skeleton.List     |
| 🇨🇳 Chinese |    Skeleton.List       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/skeleton/demo/element.md](https://github.com/vikasprashar99/ant-design/blob/skeleton.list-API/components/skeleton/demo/element.md)

### CODE REVIEW

**Comment by fireairforce** [comment]
Can you provide a picture to show the API?

**Comment by Rustin-Liu** [comment]
Please send this PR to features branch.

**Comment by Rustin-Liu** [comment]
> 
> 
> Can you provide a picture to show the API?

You can preview it on https://preview-24820-ant-design.surge.sh/components/skeleton/.

image

**Comment by xrkffgg** [comment]
可以改成 feature 的，不用新开的。

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap [hunk 1]
```diff
@@ -25192,8 +25192,8 @@ exports[`ConfigProvider components Tabs configProvider 1`] = `
         style="transform:translate(0px, 0px)"
       >
         <button
-          aria-selected="false"
-          class="config-tabs-tab"
+          aria-selected="true"
```
Comment (Rustin-Liu) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap:25195]: The CI broken. I think there is no reason to do this change. Please make sure your branch checkout from the antd's latest feature branch.

And delete your old `node_modules` folder and reinstall it, then you can retry npm test and fix the issue.

**Review by 0xPoe — CHANGES_REQUESTED** [review]

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap (hunk 1 above)
Comment (vikasprashar99) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap:25195]: okay i will do it

**Comment by Rustin-Liu** [comment]
The CI still broken. I think you test snap still have some problem.

**Comment by vikasprashar99** [comment]
@Rustin-Liu Should i open a new pull request without changing any of the tests?

**Comment by Rustin-Liu** [comment]
> 
> 
> @Rustin-Liu Should i open a new pull request without changing any of the tests?

I think you should use `npm run test -- -u` to update these snaps. If you on the right branch and have correct node_modules, I think it will restored as before.

**Comment by vikasprashar99** [comment]
@Rustin-Liu Please take a look

File: components/skeleton/List.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const renderSkeletonList = ({ getPrefixCls }: ConfigConsumerProps) => {
+    const { prefixCls: customizePrefixCls, className, active } = props;
+    const prefixCls = getPrefixCls('skeleton', customizePrefixCls);
+    const otherProps = omit(props, ['prefixCls']);
+    const cls = classNames(prefixCls, className, `${prefixCls}-element`, {
+      [`${prefixCls}-active`]: active,
+    });
+    return (
+      <div className={cls}>
+        <Element prefixCls={`${prefixCls}-list`} {...otherProps} />
```
Comment (Rustin-Liu) [inline components/skeleton/List.tsx:21]: I think if you want to use Element you should add this css class into css file.

For now, your implement is a empty span. 
image

You can preview it by `npm start`.

**Review by 0xPoe — CHANGES_REQUESTED** [review]

**Comment by fireairforce** [comment]
I think you should to implement a Component like Skeleton.Button or others but not just to exposed a api.

File: components/skeleton/List.tsx (hunk 2 above)
Comment (vikasprashar99) [reply components/skeleton/List.tsx:21]: Added css

**Comment by vikasprashar99** [comment]
> I think you should implement a Component like Skeleton.Button or others but not just to exposed a api.

please check my latest commit and tell me if any issues

File: components/skeleton/List.tsx (hunk 2 above)
Comment (Rustin-Liu) [reply components/skeleton/List.tsx:21]: I think the SkeletonList  should look like List component. It seems your impl only one element.

You can refers List component API and shape.

File: components/skeleton/style/index.less [hunk 3]
```diff
@@ -168,6 +175,24 @@
...
 }
 
+// List
+.skeleton-element-list() {
+  display: inline-block;
+  vertical-align: top;
+  background: @skeleton-color;
+  border-radius: @border-radius-base;
+
+  .skeleton-element-list-size(@btn-height-base);
```
Comment (Rustin-Liu) [inline components/skeleton/style/index.less:185]: Please don't use btn's style. You can use List component const or you define it yourself.

**Review by 0xPoe — CHANGES_REQUESTED** [review]

File: components/skeleton/style/index.less (hunk 3 above)
Comment (vikasprashar99) [reply components/skeleton/style/index.less:185]: okay will do it

Comment (vikasprashar99) [reply components/skeleton/style/index.less:185]: @Rustin-Liu I have use the same CSS used in List component?

Comment (Rustin-Liu) [reply components/skeleton/style/index.less:185]: Yes

**Comment by Rustin-Liu** [comment]
Could you please resolve the conflicts? Thanks!

**Comment by vikasprashar99** [comment]
image
@Rustin-Liu Does this looks Fine?

**Comment by Rustin-Liu** [comment]
> Does this looks Fine?

I think maybe we can make it use the same width.

Because we already have a paragraph look like this.

@afc163 What do you think?

By the way, the PR still conflicting,  you can not merge `master` to `feature` branch.

**Comment by vikasprashar99** [comment]
> > Does this looks Fine?
> 
> I think maybe we can make it use the same width.
> 
> Because we already have a paragraph look like this.
> 
> @afc163 What do you think?
> 
> By the way, the PR still conflicting, you can not merge `master` to `feature` branch.

Yes I will resolve the conflicts after the skeleton is final

**Comment by afc163** [comment]
A lot of conflict, please create another pull request.

**Pull request closed without merging**